libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -1335,32 +1413,29 @@ bool WaveClip::WithinPlayRegion(double t) const
 bool WaveClip::EntirelyWithinPlayRegion(double t0, double t1) const
 {
    assert(t0 <= t1);
-   return !BeforePlayRegion(t0) && !AfterPlayRegion(t1);
+   // t1 is the open end of the interval, hence it's ok if it's equal to the
+   // open end of the play region.
+   return !BeforePlayRegion(t0) && t1 <= GetPlayEndTime();
 }
 
 bool WaveClip::PartlyWithinPlayRegion(double t0, double t1) const
 {
    assert(t0 <= t1);
-   return OverlapsPlayRegion(t0, t1) && !EntirelyWithinPlayRegion(t0, t1);
+   return WithinPlayRegion(t0) != WithinPlayRegion(t1);
 }
 
 bool WaveClip::OverlapsPlayRegion(double t0, double t1) const
 {
    assert(t0 <= t1);
-   return !(BeforePlayRegion(t1) || AfterPlayRegion(t0));
-}
-
-bool WaveClip::ExtendsPlayRegion(double t0, double t1) const
-{
-   assert(t0 <= t1);
-   return (BeforePlayRegion(t0) && WithinPlayRegion(t1)) ||
-          (WithinPlayRegion(t0) && AfterPlayRegion(t1));
+   // t1 is the open end of the interval, so it must be excluded from the closed
+   // begin of the play region.
+   return t0 < GetPlayEndTime() && GetPlayStartTime() < t1;
 }
 
-bool WaveClip::ExtendsPlayRegionOnBothSides(double t0, double t1) const
+bool WaveClip::CoversEntirePlayRegion(double t0, double t1) const
 {
    assert(t0 <= t1);
-   return BeforePlayRegion(t0) && AfterPlayRegion(t1);
+   return t0 <= GetPlayStartTime() && GetPlayEndTime() <= t1;
```

Comment:
Some of my earlier comments are irrelevant now after these changes in the interval test functions